### PR TITLE
fix(util.ts): Improve cross-browser compatibility of `isElementInView` and `sideElementIsOutOfView`

### PR DESCRIPTION
**Pull request by TheRealJon**
Fixes https://github.com/patternfly/patternfly-react/issues/1451

**What**: 
- Use `getBoundingClientRect` and `Math.floor` in `isElementInView` and `sideElementIsOutOfView` util functions to improve consistency of element left and right bound values across browsers. 
- Update `util.test.js` to mock `getBoundingClientRect` function.

### CODE REVIEW

**Comment by TheRealJon** [comment]
Since I'm using `Math.floor` here, there is probably an edge case where an `element` overflows on the left or right of `container` by a fraction of a pixel and not be considered partially hidden.

File: packages/patternfly-4/react-core/src/helpers/util.ts [hunk 1]
```diff
@@ -22,6 +22,7 @@ export function debounce(this: any, func: (...args: any[]) => any, wait: number)
   };
 }
 
+// TODO this is not the correct description for this function
```
Comment (jschuler) [inline packages/patternfly-4/react-core/src/helpers/util.ts]: LGTM, can you update the descriptions?

**Comment by TheRealJon** [comment]
@jschuler I've updated the comments for the functions in util.ts

File: packages/patternfly-4/react-core/src/helpers/util.ts [hunk 2]
```diff
@@ -73,7 +76,17 @@ export function sideElementIsOutOfView(container: HTMLElement, element: HTMLElem
   // Return outcome
   return side;
 }
-/** This function returns the side the element is out of view on (right, left or both)
+
+/** Interpolates a parameterized templateString using values from a templateVars object.
```
Comment (jschuler) [inline packages/patternfly-4/react-core/src/helpers/util.ts]: Thanks for fixing this description as well

**Comment by TheRealJon** [comment]
Sorry, noticed we were also missing a unit test for that templateFill function, so I added one.

**Comment by TheRealJon** [comment]
@jschuler @tlabaj Is there a way to re-trigger TravisCI? I think it canceled the last run but didn't start a new one.

**Comment by tlabaj** [comment]
@TheRealJon I just restarted it.
